src/alloc_pool.c

```diff
@@ -435,7 +437,8 @@ void qd_dealloc(qd_alloc_type_desc_t *desc, qd_alloc_pool_t **tpool, char *p)
     assert (desc->header  == PATTERN_FRONT);
     assert (desc->trailer == PATTERN_BACK);
     assert (item->header  == PATTERN_FRONT);
-    assert (*((uint32_t*) (p + desc->total_size)) == PATTERN_BACK);
+    const uint32_t pb = PATTERN_BACK;
+    assert (memcmp(p + desc->total_size, &pb, sizeof(pb)) == 0);
```

Comment:
I imagine Coverity will scream about this. Memcpy has side-efects and it is being used in an assert.

Reply:
Uh, memcmp, never mind.